### PR TITLE
Fix unexpected keyword argument 'context'

### LINKED ISSUE #25: Python 3.10: TypeError: __add__() got an unexpected keyword argument 'context'
Running the example i'm getting this error (tested with Python3.9 and Python3.10)

```
 $ rp2_us -m fifo -o output -p crypto_example_ crypto_example.config crypto_example.ods
INFO: Country: us
INFO: Accounting Method: fifo
INFO: Configuration file: crypto_example.config
INFO: Input file: crypto_example.ods
INFO: Processing BTC
ERROR: Fatal exception occurred:
Traceback (most recent call last):
  File "/home/rndstr/venv/lib/python3.9/site-packages/rp2/rp2_main.py", line 93, in _rp2_main_internal
    input_data: InputData = parse_ods(configuration=configuration, asset=asset, input_file_handle=input_file_handle)
  File "/home/rndstr/venv/lib/python3.9/site-packages/rp2/ods_parser.py", line 128, in parse_ods
    _create_and_process_transaction(
  File "/home/rndstr/venv/lib/python3.9/site-packages/rp2/ods_parser.py", line 168, in _create_and_process_transaction
    transaction: AbstractTransaction = _create_transaction(configuration, current_table_type, internal_id, row_values)
  File "/home/rndstr/venv/lib/python3.9/site-packages/rp2/ods_parser.py", line 277, in _create_transaction
    transaction = InTransaction(**argument_pack)
  File "/home/rndstr/venv/lib/python3.9/site-packages/rp2/in_transaction.py", line 52, in __init__
    super().__init__(configuration, timestamp, asset, transaction_type, spot_price, internal_id, unique_id, notes)
  File "/home/rndstr/venv/lib/python3.9/site-packages/rp2/abstract_transaction.py", line 41, in __init__
    self.__spot_price: RP2Decimal = configuration.type_check_positive_decimal("spot_price", spot_price)
  File "/home/rndstr/venv/lib/python3.9/site-packages/rp2/configuration.py", line 276, in type_check_positive_decimal
    if result < ZERO:
  File "/home/rndstr/venv/lib/python3.9/site-packages/rp2/rp2_decimal.py", line 59, in __lt__
    return not self.__ge__(other)
  File "/home/rndstr/venv/lib/python3.9/site-packages/rp2/rp2_decimal.py", line 48, in __ge__
    return (self - other).quantize(CRYPTO_DECIMAL_MASK).__ge__(ZERO)
  File "/home/rndstr/venv/lib/python3.9/site-packages/rp2/rp2_decimal.py", line 69, in __sub__
    return RP2Decimal(Decimal.__sub__(self, other))
  File "/usr/lib/python3.9/_pydecimal.py", line 1257, in __sub__
    return self.__add__(other.copy_negate(), context=context)
TypeError: __add__() got an unexpected keyword argument 'context'
INFO: Log file: ./log/rp2_2022_04_10_15_52_55_045185.log
INFO: Generated output directory: output
INFO: Done
```

### PR DESCRIPTION
**Pull request by rndstr**
Updates signature of `RP2Decimal.__add__()` to allow the `context` param which is
being invoked at
https://github.com/python/cpython/blob/f33e2c87a83917b5139d97fd8ef7cba7223ebef5/Lib/_pydecimal.py#L1257

Fixes #25

### CODE REVIEW

**Comment by rndstr** [comment]
I presume we should change all the signatures in that file to include a context.

File: src/rp2/rp2_decimal.py [hunk 1]
```diff
@@ -34,97 +34,97 @@ class RP2Decimal(Decimal):
...
 
-    def __rtruediv__(self, other: object) -> "RP2Decimal":
+    def __rtruediv__(self, other: object, context = None) -> "RP2Decimal":
         if not isinstance(other, Decimal):
             raise RP2TypeError(f"Operand has non-Decimal value {repr(other)}")
-        return RP2Decimal(Decimal.__rtruediv__(self, other))
+        return RP2Decimal(Decimal.__rtruediv__(self, other, context))
 
-    def __rfloordiv__(self, other: object) -> "RP2Decimal":
+    def __rfloordiv__(self, other: objec, context = None) -> "RP2Decimal":
```
Comment (eprbell) [inline src/rp2/rp2_decimal.py:119]: typo: object

File: src/rp2/rp2_decimal.py [hunk 2]
```diff
@@ -34,97 +34,97 @@ class RP2Decimal(Decimal):
...
         if not isinstance(other, Decimal):
             raise RP2TypeError(f"Operand has non-Decimal value {repr(other)}")
-        return RP2Decimal(Decimal.__rfloordiv__(self, other))
+        return RP2Decimal(Decimal.__rfloordiv__(self, other, context))
 
-    def __rmod__(self, other: object) -> "RP2Decimal":
+    def __rmod__(self, other: objec, context = None) -> "RP2Decimal":
```
Comment (eprbell) [inline src/rp2/rp2_decimal.py:124]: typo: object

**Comment by eprbell** [comment]
I added Python 3.10 to the continuous integration test matrix (on Windows, Mac and Linux) and all unit tests pass: this seems related to a problem with the specific Python installation on your machine (see also my comments in #25 and #30). You may want to try uninstalling and reinstalling Python.

**Pull request closed without merging**